### PR TITLE
✨ Use go.mod for the operator

### PR DESCRIPTION
**Pull request by asalkeld**
**What this PR does / why we need it**:

This has been separated out of #4926 at the request of @alexander-demichev 

Add a go.mod to exp/operator/ to make getting the latest capi code easier (instead of rebasing).

Note: because it is now a separate module much of the top level makefile targets do not work as expected 
(the new sub-module just gets ignored).

This also includes the following changes to make this work:
- envtest has changed, so duplicate the makefile
- internal/envtest was not picking up the crds (different go mod), so duplicate it
  so we can add the path to the crds (and remove the operator from the top level envtest).
- fix some lint errors

/area operator
/cc @fabriziopandini @alexander-demichev

### CODE REVIEW

**Comment by asalkeld** [comment]
Note the api tests will aways fail at it looks like we have remove the operator API because it is in a new module..

**Review by alexander-demichev — APPROVED** [review]
/lgtm

go.mod file references a CAPI fork with required clusterctl changes, but once they are merged to master we should remove the replace.

File: exp/operator/go.mod [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	github.com/onsi/gomega v1.14.0
+	github.com/pkg/errors v0.9.1
+	github.com/spf13/pflag v1.0.5
+	k8s.io/api v0.21.3
+	k8s.io/apiextensions-apiserver v0.21.3
+	k8s.io/apimachinery v0.21.3
+	k8s.io/client-go v0.21.3
+	k8s.io/klog/v2 v2.9.0
+	sigs.k8s.io/cluster-api v0.0.0-00010101000000-000000000000
+	sigs.k8s.io/controller-runtime v0.9.6
```
Comment (fabriziopandini) [inline exp/operator/go.mod:17]: CAPI is on v0.9.7 now, should we align?

Comment (asalkeld) [reply exp/operator/go.mod]: thanks, i have justed sync'd the versions with master.

**Comment by alexander-demichev** [comment]
@asalkeld  Can cluster-api be pointed main branch instead of a fork?

**Comment by asalkeld** [comment]
> @asalkeld Can cluster-api be pointed to main branch instead of a fork?

@alexander-demichev I just opened this PR https://github.com/kubernetes-sigs/cluster-api/pull/5300 with the last 2 small things.
The we can use it from main.

**Comment by enxebre** [comment]
@asalkeld @alexander-demichev  do we want to close this now the operator is moving into its own repo?

**Pull request closed without merging**